### PR TITLE
Line numbers

### PR DESCRIPTION
**Pull request by rlisahuang**
Separating changes for line numbers out to the new branch. Closing #96 .

### CODE REVIEW

File: website/src/components/TutorialComponents/CodeBlock.tsx [hunk 1]
```diff
@@ -36,13 +38,24 @@ export function GithubDiscussionBtn({ repo }) {
...
+function CodeEditor(props: {
+    className: string,
+    code: string,
+    disabled: boolean,
+    language: string,
+    onChange: (code: string) => void,
+    prism: typeof Prism,
+    style?: CSSProperties,
+    theme: PrismTheme,
+    githubRepo: string | undefined,
```
Comment (pelikhan) [inline website/src/components/TutorialComponents/CodeBlock.tsx:50]: Or use ? For optimal fields 


    githubRepo?: string

File: website/src/components/TutorialComponents/CodeBlock.tsx [hunk 2]
```diff
@@ -36,13 +38,24 @@ export function GithubDiscussionBtn({ repo }) {
...
+    showLineNumbers: boolean,
+}) {
     const editorRef = useRef(null);
     const [code, setCode] = useState(props.code || "");
 
     useEffect(() => {
-        setCode(props.code);
-    }, [props.code]);
+        setCode(code);
+    }, [code]);
```
Comment (pelikhan) [inline website/src/components/TutorialComponents/CodeBlock.tsx:58]: What is the purpose of this effect?

File: website/src/remark/render-code-blocks.mjs [hunk 3]
```diff
@@ -167,16 +167,23 @@ export default function plugin() {
...
 
-            const skipRegex = new RegExp(/(no-build)|(ignore-errors)/g);
-            const skipErr = meta && meta.match(skipRegex) !== null;
-            const editableRegex = new RegExp(/(always-editable)/g);
-            const alwaysEditable = meta && meta.match(editableRegex) !== null;
+            const skipRegex = /(no-build)|(ignore-errors)/;
+            const skipErr = skipRegex.test(meta);
+            const editableRegex = /(always-editable)/;
+            const alwaysEditable = editableRegex.test(meta);
+            const lineNumRegex = /(showLineNumbers)/i;
```
Comment (pelikhan) [inline website/src/remark/render-code-blocks.mjs:174]: Pixies camel case and dashed naming schemes... pick one :)

**Review by pelikhan — APPROVED** [review]

File: website/src/components/TutorialComponents/CodeBlock.tsx (hunk 1 above)
Comment (rlisahuang) [reply website/src/components/TutorialComponents/CodeBlock.tsx:50]: The field gets passed in all the time from the remark plugin, it's just that the value can be `undefined`.

File: website/src/components/TutorialComponents/CodeBlock.tsx (hunk 2 above)
Comment (rlisahuang) [reply website/src/components/TutorialComponents/CodeBlock.tsx:58]: Wrong edits. Should be `props.code` not the local variable `code`.

Comment (rlisahuang) [reply website/src/components/TutorialComponents/CodeBlock.tsx]: The side effect set the value of `code` variable to `props.code` again if `props.code` changes.

File: website/src/remark/render-code-blocks.mjs (hunk 3 above)
Comment (rlisahuang) [reply website/src/remark/render-code-blocks.mjs]: dashed naming it is!

**Review by rlisahuang — COMMENTED** [review]
Comments from the previous review were addressed.

**Pull request merged**